STDatabase/src/commonTest/kotlin/com/infomaniak/multiplatform_swisstransfer/database/TransferControllerTest.kt

```diff
@@ -151,6 +151,32 @@ class TransferControllerTest {
         assertEquals(1, transferController.getOrphanContainers().count(), "We should have an orphan container")
     }
 
+    @Test
+    fun ensureTransferStatusIsPreservedOnUpdateIfDownloadCounterCreditIsMoreThanZero() = runTest {
+        val transfer = DummyTransfer.transfer3
+
+        transferController.insert(transfer, TransferDirection.SENT, transfer.password)
+        val realmTransfer = transferController.getTransfer(transfer.linkUUID)!!
+
+        transferController.update(realmTransfer)
+        val realmTransfer2 = transferController.getTransfer(transfer.linkUUID)!!
+
+        assertEquals(transfer.transferStatus, realmTransfer2.transferStatus)
+    }
+
+    @Test
+    fun ensureTransferStatusIsExpiredDownloadQuotaOnUpdateIfDownloadCounterCreditIsZero() = runTest {
+        val transfer = DummyTransfer.transfer4
+
+        transferController.insert(transfer, TransferDirection.SENT, transfer.password)
+        val realmTransfer = transferController.getTransfer(transfer.linkUUID)!!
+
+        transferController.update(realmTransfer)
+        val realmTransfer2 = transferController.getTransfer(transfer.linkUUID)!!
+
+        assertNotEquals(transfer.transferStatus, realmTransfer2.transferStatus)
```

Comment:
And check that the status is indeed different from the expected one, and that at the end you have the expected status.
Never trust data that can change at any time